### PR TITLE
Fix#467 bypass value to validate message

### PR DESCRIPTION
**Pull request by fabiopaiva**

### CODE REVIEW

**Comment by Ocramius** [comment]
@fabiopaiva requires a test case

**Comment by fabiopaiva** [comment]
<?php

namespace Application\Form;

use Zend\Form\Form;
use Zend\InputFilter\InputFilter;
use Application\Entity\UnidadeMedicao as UnidadeMedicaoEntity;
use DoctrineModule\Stdlib\Hydrator\DoctrineObject as DoctrineHydrator;
use Doctrine\ORM\EntityManager;

class UnidadeMedicao extends Form {

    public function __construct(EntityManager $em) {
        parent::__construct('unidade-medicao');

        $this
                ->setAttribute('class', 'form-horizontal')
                ->setHydrator(new DoctrineHydrator($em))
                ->setObject(new UnidadeMedicaoEntity())
                ->add(array(
                    'type' => 'hidden',
                    'name' => 'id'
                ))
                ->add(array(
                    'name' => 'nome',
                    'options' => array(
                        'label' => 'Abreviatura'
                    ),
                    'attributes' => array(
                        'class' => 'form-control input-sm',
                    )
                        )
                )
                ->add(array(
                    'name' => 'descricao',
                    'options' => array(
                        'label' => 'Descrição'
                    ),
                    'attributes' => array(
                        'class' => 'form-control input-sm',
                    )
                        )
                )
        ;

        $filter = new InputFilter();
        $filter
                ->add(array(
                    'name' => 'nome',
                    'required' => true,
                    'validators' => array(
                        array(
                            'name' => '\DoctrineModule\Validator\UniqueObject',
                            'options' => array(
                                'use_context' => true,
                                'object_repository' => $em
                                ->getRepository('Application\Entity\UnidadeMedicao'),
                                'object_manager' => $em,
                                'fields' => 'nome',
                                'messages' => array(
                                    'objectNotUnique' => 'Esta abreviatura já está cadastrado em nossa base de dados %value%'
                                )
                            )
                        )
                    )
                ))
                ->add(array(
                    'name' => 'descricao',
                    'required' => true
                ))
        ;
        $this->setInputFilter($filter);
    }

}


Result:

print

**Comment by Ocramius** [comment]
Test case as in unit-test ;-)

**Comment by fabiopaiva** [comment]
Sorry, i don't know how to do. 
:(

**Comment by Ocramius** [comment]
@fabaopaiva look at the existing tests ;-)

**Comment by fabiopaiva** [comment]
@Ocramius 
Created tests, its look ok?

File: .travis.yml [hunk 1]
```diff
@@ -1,7 +1,6 @@
 language: php
 
 php:
-  - 5.3
```
Comment (Ocramius) [inline .travis.yml]: These changes are unrelated to the PR

File: tests/DoctrineModuleTest/Validator/NoObjectExistsTest.php [hunk 2]
```diff
@@ -59,4 +59,27 @@ public function testCannotValidateWithAvailableObjectInRepository()
...
+    public function testErrorMessageIsStringInsteadArray()
+    {
+        $repository = $this->getMock('Doctrine\Common\Persistence\ObjectRepository');
+        $repository
+            ->expects($this->once())
+            ->method('findOneBy')
+            ->will($this->returnValue(new stdClass()));
+        
+        $validator = $this->getMock(
+            'DoctrineModule\Validator\NoObjectExists',
```
Comment (Ocramius) [inline tests/DoctrineModuleTest/Validator/NoObjectExistsTest.php]: Never mock the system under test

File: tests/DoctrineModuleTest/Validator/ObjectExistsTest.php [hunk 3]
```diff
@@ -184,4 +184,29 @@ public function testWillNotValidateOnFieldKeysMismatch()
...
+    public function testErrorMessageIsStringInsteadArray()
+    {
+        $repository = $this->getMock('Doctrine\Common\Persistence\ObjectRepository');
+        $repository
+            ->expects($this->once())
+            ->method('findOneBy')
+            ->will($this->returnValue(null));
+
+        $validator = $this->getMock(
+            'DoctrineModule\Validator\ObjectExists',
```
Comment (Ocramius) [inline tests/DoctrineModuleTest/Validator/ObjectExistsTest.php]: Don't mock the system under test

File: tests/DoctrineModuleTest/Validator/UniqueObjectTest.php [hunk 4]
```diff
@@ -398,4 +398,60 @@ public function testCanFetchIdentifierFromObjectContext()
...
+            ->method('getClassName')
+            ->will($this->returnValue('stdClass'));
+        $repository
+            ->expects($this->once())
+            ->method('findOneBy')
+            ->with(array('matchKey' => 'matchValue'))
+            ->will($this->returnValue($match));
+
+        $validator = $this->getMock(
+            'DoctrineModule\Validator\UniqueObject',
```
Comment (Ocramius) [inline tests/DoctrineModuleTest/Validator/UniqueObjectTest.php]: Don't mock the system under test

File: tests/DoctrineModuleTest/Validator/NoObjectExistsTest.php (hunk 2 above)
Comment (fabiopaiva) [inline tests/DoctrineModuleTest/Validator/NoObjectExistsTest.php]: How to check the value passed in the error method?
Method isValid return bool 
https://github.com/doctrine/DoctrineModule/blob/master/src/DoctrineModule/Validator/UniqueObject.php#L121

and method error is in a parent class (AbstractValidator)

how to check the value passed in method is ok?

i tried mock AbstractValidator but don't work.

Comment (Ocramius) [inline tests/DoctrineModuleTest/Validator/NoObjectExistsTest.php]: @fabiopaiva just assert on the final error message

**Comment by fabiopaiva** [comment]
@Ocramius its ok?

**Comment by gianarb** [comment]
I have already merged #479 
Close this PR 

Thanks!

**Comment by fabiopaiva** [comment]
You´re welcome

**Comment by gianarb** [comment]
:) Thanks
